### PR TITLE
Fix more test that fail due to cypher deprecation removals

### PR DESCRIPTION
**Pull request by gjmwoods**

### CODE REVIEW

**Review by fbiville — APPROVED** [review]

File: driver/src/test/java/org/neo4j/driver/integration/ConnectionHandlingIT.java [hunk 1]
```diff
@@ -254,7 +254,7 @@ void connectionUsedForTransactionReturnedToThePoolWhenTransactionFailsToCommitte
     {
         try ( Session session = driver.session() )
         {
-            session.run( "CREATE CONSTRAINT ON (book:Library) ASSERT exists(book.isbn)" );
+            session.run( "CREATE CONSTRAINT FOR (book:Library) REQUIRE book.isbn IS NOT NULL" );
```
Comment (fbiville) [inline driver/src/test/java/org/neo4j/driver/integration/ConnectionHandlingIT.java]: this needs to depend on the tested neo4j version
(this syntax is valid with 4.4 and 5.0 but not earlier)

**Review by fbiville — CHANGES_REQUESTED** [review]

File: driver/src/test/java/org/neo4j/driver/util/DatabaseExtension.java [hunk 2]
```diff
@@ -174,13 +160,15 @@ public void stopDb()
         runner.stopNeo4j();
     }
 
-    public int getMajorVersion()
+    public boolean isNeo4j43OrEarlier()
     {
         try ( Session session = driver().session() )
         {
             String protocolVersion = session.readTransaction( tx -> tx.run( "RETURN 1" ).consume().server().protocolVersion() );
```
Comment (michael-simons) [inline driver/src/test/java/org/neo4j/driver/util/DatabaseExtension.java]: I would rather use 

```
CALL dbms.components() YIELD versions, edition 
RETURN 'Neo4j/' + versions[0] AS version, edition
```

bolt protocol depends on what server and client negotiated.

**Review by michael-simons — CHANGES_REQUESTED** [review]

File: driver/src/test/java/org/neo4j/driver/util/DatabaseExtension.java [hunk 3]
```diff
@@ -174,13 +160,16 @@ public void stopDb()
         runner.stopNeo4j();
     }
 
-    public int getMajorVersion()
+    public boolean isNeo4j43OrEarlier()
     {
         try ( Session session = driver().session() )
         {
-            String protocolVersion = session.readTransaction( tx -> tx.run( "RETURN 1" ).consume().server().protocolVersion() );
+            String protocolVersion = session.readTransaction( tx -> tx.run( "CALL dbms.components() YIELD versions " +
```
Comment (michael-simons) [inline driver/src/test/java/org/neo4j/driver/util/DatabaseExtension.java]:
```suggestion
            String neo4jVersion = session.readTransaction( tx -> tx.run( "CALL dbms.components() YIELD versions " +
```

File: driver/src/test/java/org/neo4j/driver/util/DatabaseExtension.java [hunk 4]
```diff
@@ -174,13 +160,16 @@ public void stopDb()
...
+                                                                            "RETURN versions[0] AS version" ).single().get( "version" ).asString() );
             String[] versions = protocolVersion.split( "\\." );
```
Comment (michael-simons) [inline driver/src/test/java/org/neo4j/driver/util/DatabaseExtension.java]:
```suggestion
            String[] versions = neo4jVersion.split( "\\." );
```

**Review by michael-simons — CHANGES_REQUESTED** [review]
Sorry for the nitpicking on wording.

**Review by michael-simons — APPROVED** [review]
Thanks.

File: testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/SessionWriteTransaction.java [hunk 5]
```diff
@@ -123,6 +123,10 @@ public TestkitResponse process( TestkitState testkitState )
                 {
                     workThrowable = workThrowable.getCause();
                 }
+                if ( workThrowable instanceof FrontendError )
```
Comment (injectives) [inline testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/SessionWriteTransaction.java]: Is this duplicate of the block below?

File: testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/SessionReadTransaction.java [hunk 6]
```diff
@@ -120,6 +120,10 @@ public TestkitResponse process( TestkitState testkitState )
                 {
                     workThrowable = workThrowable.getCause();
                 }
+                if ( workThrowable instanceof FrontendError )
```
Comment (injectives) [inline testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/SessionReadTransaction.java]: Is this duplicate of the block below?

**Pull request merged**